### PR TITLE
Basic layout for writing tests using jest

### PR DESCRIPTION
**Pull request by ganeshpatro321**
Tests can be written in files in the form <filename>.test.js

I have created a dummy test to show the layout in the file app.test.js

you can run "yarn test" to run the test files. 

@domoritz

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  presets: ["@babel/preset-env", "@babel/preset-react"]
```
Comment (domoritz) [inline babel.config.js]: Just curious, why do we need babel when we have typescript?

File: src/app.test.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+test("Fake Test", () => {
```
Comment (domoritz) [inline src/app.test.js]: Let's write the tests in typescript. See vega-lite for how you could set it up.

**Comment by siddhant1** [comment]
This setup will remove tree-shaking from the app , also configure jest to include local storage since we are using it , also I prefer react-testing-library or enzyme for testing . I have all the setup ready from the previous app , should I make a pr?

**Comment by domoritz** [comment]
My requirements:
* As few dependencies as possible
* Support Typescript
* Support testing the app
* Easy to use. 

If you check those boxes, yes, please send a PR!

**Comment by siddhant1** [comment]
The app can be tested only with jest but that'll be a hassle to maintain , also the following pr will make tree-shaking disabled which will slow down the app. Also no support for local storage too ,  I will have a look to improve this .

**Comment by siddhant1** [comment]
There is a global babel config which will disable tree shaking in general (because it will transpile esm)

**Comment by ganeshpatro321** [comment]
> My requirements:
> 
>     * As few dependencies as possible
> 
>     * Support Typescript
> 
>     * Support testing the app
> 
>     * Easy to use.
> 
> 
> If you check those boxes, yes, please send a PR!

I used jest because it is light weight and easy to use. It provides features like snapshot testing which will make make component testing easy.

I was not aware of the breaking of tree-shaking and searched the web for the same, but couldn't find anything for the same.

@domoritz I will recheck whether we need babel or not for typescript, I followed official jest docs while setting up the configuration and ended up here.

Please let me know if you have any further doubts.
@siddhant1

**Comment by domoritz** [comment]
Okay, since this is a bit more work, let's close this PR. Please open a new one when it's ready.

**Pull request closed without merging**